components/brave_federated/public/interfaces/brave_federated.mojom

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+module brave_federated.mojom;
+
+enum CovariateType {
+  kAdNotificationWasClicked,
+  kAdNotificationLocaleCountryAtTimeOfServing,
+  kAdNotificationImpressionServedAt,
+  kAdNotificationNumberOfTabsOpenedInPast30Minutes
+};
+
+enum DataType {
+  kBool,
+  kDouble,
+  kInt,
+  kString
+};
+
+struct Covariate {
+  DataType data_type;
+  CovariateType covariate_type;
+  string value;
+};
+
+struct TrainingCovariates {
```

Comment:
Nit: unrelated to this PR but this struct could just be named training as we then use training.covariates

Reply:
created a ticket